server/utils/queryHelpers/communitySanitize.js

```diff
@@ -35,5 +23,39 @@ export default (communityData, locationData, loginData, scopeData) => {
 				page: availablePages[collection.pageId],
 			};
 		});
+
+	cleanedData.pages = cleanedData.pages.filter((item) => {
+		const pageCollection = cleanedData.collections.find(
+			({ pageId }) => typeof pageId === 'string' && pageId === item.id,
+		);
+
+		// If the page has a collection, check if the user has manage permission
+		// to that collection. If so, include the page.
+		const hasPageCollectionManageAccess = pageCollection
+			? Boolean(
+					pageCollection.members.find(
+						(member) =>
+							member.userId === loginData.id && member.permissions === 'manage',
```

Comment:
We also want to check for the `admin` permission here

...really what we want is to break a bunch of utility functions out of `scopeGet.js`, which handles all of these edge cases correctly, but in mega-functions which would be convenient to access as a collection of smaller functions.